features/ouders.feature

```diff
@@ -2,113 +2,347 @@
+    | 3         |                    |               |                       |                             |                                                    |
+    En in categorie 3 komt alleen groep 81 en groep 85 voor
+    Als personen wordt gezocht met de volgende parameters
+    | naam                | waarde                          |
+    | type                | RaadpleegMetBurgerservicenummer |
+    | burgerservicenummer | 555550001                       |
+    | fields              | ouders                          |
+    Dan bevat de persoon alleen ouders met de volgende gegevens
+    | naam                                    | waarde   |
+    | ouderAanduiding                         | ouder1   |
+    En heeft de ouder met ouderAanduiding 'ouder1' de volgende gegevens
+    | naam                                    | waarde   |
+    | type                                    | GbaOuder |
+    | ouderAanduiding                         | ouder1   |
+    | datumIngangFamilierechtelijkeBetrekking | 00000000 |
+    En de ouder met ouderAanduiding 'ouder1' heeft GEEN 'geslachtsaanduiding' gegevens
+    En de ouder met ouderAanduiding 'ouder1' heeft GEEN 'geboorte' gegevens
+    En heeft de ouder met ouderAanduiding 'ouder1' alleen de volgende 'naam' gegevens
+    | naam          | waarde |
+    | geslachtsnaam | .      |
+
+  @gba
+  Scenario: met fields vragen om gegevens zonder waarde
+    Gegeven de persoon met burgerservienummer 555550001 heeft de volgende ouders in de registratie
+    | Categorie | Voornamen (02.10)  | Geslachtsnaam | Geboortedatum (03.10) | Geslachtsaanduiding (04.10) | Datum ingang familierechtelijke betrekking (62.10) |
+    | 2         |                    | Jansen        |                       | V                           | 20190614                                           |
+    | 3         |                    |               |                       |                             |                                                    |
+    En in categorie 3 komt alleen groep 81 en groep 85 voor
+    Als personen wordt gezocht met de volgende parameters
+    | naam                | waarde                                 |
+    | type                | RaadpleegMetBurgerservicenummer        |
+    | burgerservicenummer | 555550001                              |
+    | fields              | ouders.voornamen,ouders.geboorte.datum |
+    Dan heeft de persoon met burgerservicenummer '555550001' exact 1 ouders
+    En heeft de ouder GEEN gegevens
+
+@proxy
+Rule: Wanneer de geslachtsnaam van de ouder onbekend is, wordt de ouder geleverd met type "OnbekendOuder" en indicatieOnbekend met waarde true
+  - Dit is het geval wanneer geslachtsnaam dan de standaardwaarde "." heeft
+  - Wanneer van de ouder wel gegevens geregistreerd zijn, maar geen van de met fields gevraagde gegevens heeft een waarde, dan is het type "Ouder" en wordt indicatieOnbekend NIET opgenomen
```

Comment:
Ook al heeft geslachtsnaam de standaardwaarde '.'?

Reply:
Ja juist wanneer geslachtsnaam de standaardwaarde punt heeft, is het een onbekendOuder

Reply:
Ik ben denk ik niet duidelijk. Het gaat mij om de situatie die bij het 2e streepje wordt omschreven, waarin:
* er wel gegevens van de ouder geregistreerd zijn
* en de met fields gevraagde gegevens geen waarde hebben

in dat geval zeg je dat het type 'Ouder' moet zijn. 

Dat geldt denk ik echter niet als de geslachtsnaam de standaardwaarde '.' heeft of leeg is.

Volgens mij moet de tekst bij het 2e streepje dan ook luiden:

> - Wanneer van de ouder wel gegevens geregistreerd zijn waaronder een geslachtsnaam, maar geen van de met fields gevraagde gegevens heeft een waarde, dan is het type "Ouder" en wordt indicatieOnbekend NIET opgenomen

Reply:
dat is wel waar, maar ik denk dat de zin daar niet duidelijker van wordt. Daarom is er hiervoor een scenario opgenomen op regel 321 die dit illustreert